drivers/ws2812.h

```diff
@@ -56,6 +56,12 @@
 #    define WS2812_TRST_US 280
 #endif
 
+#if defined(RGBLED_NUM)
+#    define WS2812_LED_NUM RGBLED_NUM
+#elif defined(DRIVER_LED_TOTAL)
+#    define WS2812_LED_NUM DRIVER_LED_TOTAL
+#endif
```

Comment:
Maybe this should be wrapped inside `#if !defined(WS2812_LED_NUM)`, so that it would be possible to define `WS2812_LED_NUM` externally? Otherwise a hybrid configuration with some WS2812 LEDs and some more LEDs on another controller chip (which is currently possible to implement using a custom RGB Matrix driver — e.g., see `kprepublic/bm60hsrgb/rev2`) would need to keep defining `RGBLED_NUM` even when using RGBLIGHT is not intended.

Reply:
@sigprof sounds more like a new feature to have multi-controller RGB set ups.
`WS2812_LED_NUM` should not be defined externally: for RGBLED, there is `RGBLED_NUM` and for RGBMATRIX, there is `DRIVER_LED_TOTAL` -- both fully documented for their use case.

Introducing `WS2812_LED_NUM` is only meant as an internal abstraction to reuse the same driver for both lighting subsystems. I could see it wrapped with an `!defined` to make it easier for custom implementations, though the kprepublic/bm60hsrgb/rev2` you linked, is using a custom buffer and driver implementation anyway.

Reply:
So you code cleans up the cases like `kprepublic/bm60hsrgb/rev1` (`RGB_MATRIX_DRIVER = WS2812`) — although that particular board would still need to define `RGBLED_NUM` if `RGBLIGHT_ENABLE` is set instead of `RGB_MATRIX_ENABLE` (there are a couple of other boards which also have an option to use RGBLIGHT for per-key WS2812 LEDs), and does nothing for hybrid configs with custom drivers (although the existing situation for such hybrid board is not ideal anyway, because the ISSI drivers have arrays with sizes derived from `DRIVER_LED_TOTAL`, even when not all of those LEDs are controlled by those drivers). This might be acceptable (existing boards like that would have `RGBLED_NUM` defined anyway, and your change uses `RGBLED_NUM` in preference to `DRIVER_LED_TOTAL` if both are defined).

Reply:
I'm not sure I got your point. Let me try to re-frame the problem and what I'm trying to accomplish in this PR:

Taking https://docs.qmk.fm/#/feature_rgb_matrix?id=ws2812:
It only specifies two additional defines to set when using the WS2812 driver for RGB_MATRIX.
Trying to compile a keyboard like this will 100% cause an error due to missing `RGBLED_NUM`.

This PR adds a check to use the `DRIVER_LED_TOTAL` (which *is* specified in the RGB MATRIX docs) and assign it to `RGBLED_NUM` so it uses the same driver code and feeds it the correct value.

If a user has RGBLIGHT, then nothing changes.

If a user has RGBLIGHT and RGB_MATRIX, then nothing changes.

If a user already has the required but not-documented `RGBLED_NUM` and the documented `DRIVER_LED_TOTAL`, then nothing changes.

If a user follows the docs and only defined `DRIVER_LED_TOTAL`, then it will just work.